pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -32,10 +34,58 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.images = images
         self.exclude_arches = [] if not exclude_arches else exclude_arches.replace(',', ' ').split()
         self.skip_multiarch_payload = skip_multiarch_payload
+        self.triggered_from_gen_assembly = triggered_from_gen_assembly
+        self.build_number = build_number
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
 
+    async def comment_on_pr(self):
+        """
+        Comment the link to this jenkins build on the assembly PR if it was triggered automatically
+        """
+        if self.triggered_from_gen_assembly:
+            self.logger.info("Found that build-sync was triggered by gen-assembly automatically")
+
+            owner = "openshift-eng"
+            repository = "ocp-build-data"
+            branch = self.doozer_data_gitref
+            token = os.environ.get('GITHUB_TOKEN')
+
+            api = GhApi(owner=owner, repo=repository, token=token)
+
+            # Head needs to have the repo name prepended for GhApi to fetch the correct one
+            head = f"{owner}:{branch}"
+            # Find our assembly PR.
+            prs = api.pulls.list(head=head, state="open")
+
+            if len(prs) != 1:
+                self.logger.error(f"{len(prs)} PR(s) were found from the auto generated branch {branch}. We need only 1.")
+                return
+
+            pr_number = prs[0]["number"]
+
+            text_body = f"Build sync job [{self.build_number}](https://saml.buildvm.openshift.eng.bos.redhat.com:8888" \
+                        f"/job/aos-cd-builds/job/build%252Fbuild-sync/{self.build_number}) has been triggered"
+
+            if self.runtime.dry_run:
+                self.logger.warning(f"[DRY RUN] Would have commented on PR {owner}/{repository}/pull/{pr_number} with the message: \n {text_body}")
+                return
+
+            # https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#create-an-issue-comment
+            # PR is an issue as far as  GitHub API is concerned
+            api.issues.create_comment(issue_number=pr_number, body=text_body)
+
+        else:
+            self.logger.info("Not commenting on PR since its not triggered from gen assembly automatically")
+
     async def run(self):
+        # Comment on PR if triggered from gen assembly
+        # Keeping in try-except so that job doesn't fail because of any error here
+        try:
+            await self.comment_on_pr()
+        except Exception as e:
+            self.logger.debug(f"Error commenting to PR: {e}")
```

Comment:
`debug` level is too low. We usually don't run in debug level in the pipeline.

Reply:
Understood! Updated!